exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ScanBatch.java

```diff
@@ -196,8 +196,8 @@ public IterOutcome next() {
             releaseAssets();
             done = true;  // have any future call to next() return NONE
 
-            if (mutator.isNewSchema()) {
-              // This last reader has a new schema (e.g., we have a zero-row
+            if (mutator.isNewTopLevelSchema()) {
```

Comment:
This method was added for the purpose to make count on map even when deeper schema (child fields of map) was changed. But now I  noticed that even simple select query on Map with different deeper (child) schema doesn't work. That's why no need in this method.
I returned to isNewShema method.